packages/react-next/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -0,0 +1,149 @@
+
+  useFocusRects(rootRef);
+  useDebugWarning(props);
+  useComponentRef(props, isChecked, isIndeterminate, checkBox);
+
+  const id = useId('checkbox-', props.id);
+  const classNames: { [key in keyof ICheckboxStyles]: string } = getClassNames(styles!, {
+    theme: theme!,
+    className,
+    disabled,
+    indeterminate: isIndeterminate,
+    checked: isChecked,
+    reversed: boxSide !== 'start',
+    isUsingCustomLabelRender: !!props.onRenderLabel,
+  });
+
+  const onRenderLabel = (): JSX.Element | null => {
+    return label ? (
+      <span aria-hidden="true" className={classNames.text} title={title}>
+        {label}
+      </span>
+    ) : null;
+  };
+
+  const _onChange = (ev: React.ChangeEvent<HTMLElement>): void => {
+    if (!isIndeterminate) {
+      setIsChecked(!isChecked, ev);
+    } else {
+      // If indeterminate, clicking the checkbox *only* removes the indeterminate state (or if
+      // controlled, lets the consumer know to change it by calling onChange). It doesn't
+      // change the checked state.
+      setIsChecked(!!isChecked, ev);
+      setIsIndeterminate(false);
+    }
+  };
+
+  return (
+    <KeytipData keytipProps={keytipProps} disabled={disabled}>
+      {(keytipAttributes: any): JSX.Element => (
+        <div className={classNames.root} title={title} ref={mergedRootRefs}>
+          <input
+            type="checkbox"
+            {...inputProps}
+            data-ktp-execute-target={keytipAttributes['data-ktp-execute-target']}
+            checked={!!isChecked}
+            disabled={disabled}
+            className={classNames.input}
+            ref={checkBox}
+            name={name}
+            id={id}
+            title={title}
+            onChange={_onChange}
+            onFocus={inputProps?.onFocus}
+            onBlur={inputProps?.onBlur}
+            aria-disabled={disabled}
+            aria-label={ariaLabel || label}
+            aria-labelledby={ariaLabelledBy}
+            aria-describedby={mergeAriaAttributeValues(ariaDescribedBy, keytipAttributes['aria-describedby'])}
+            aria-posinset={ariaPositionInSet}
+            aria-setsize={ariaSetSize}
+            aria-checked={isIndeterminate ? 'mixed' : isChecked ? 'true' : 'false'}
+          />
+          <label className={classNames.label} htmlFor={id}>
+            <div className={classNames.checkbox} data-ktp-target={keytipAttributes['data-ktp-target']}>
+              <Icon iconName="CheckMark" {...checkmarkIconProps} className={classNames.checkmark} />
+            </div>
+            {(props.onRenderLabel || onRenderLabel)(props, onRenderLabel)}
+          </label>
+        </div>
+      )}
+    </KeytipData>
+  );
+});
+
+CheckboxBase.displayName = 'CheckboxBase';
+
+function useDebugWarning(props: ICheckboxProps) {
+  if (process.env.NODE_ENV !== 'production') {
+    // This is a build-time conditional that will be constant at runtime
+    // tslint:disable-next-line:react-hooks-nesting
+    React.useEffect(() => {
```

Comment:
you need to call hook at top level: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level
@ecraig12345  we really need to get react hook lint rules enabled. do you know what's still blocking?

Reply:
oh sorry i just saw the comments

Reply:
This change was made in response to @dzearing's c[omment](https://github.com/microsoft/fluentui/pull/12614#discussion_r416918566). As stated in the comment, it's constant conditional at run-time, so it doesn't run afoul of the Rules of Hooks.

https://github.com/Gelio/tslint-react-hooks has the `react-hooks-nesting` rule I'm pre-emptively disabling with this comment, which we have enabled in OWA. Unfortunately, there's currently no equivalent of the `react-hooks/exhaustive-deps` ESlint rule implemented for TSLint (that I'm aware of)